packages/api/src/command/medical/cq-directory/parse-cq-organization.ts

```diff
@@ -0,0 +1,62 @@
+import { Contained } from "@metriport/carequality-sdk/models/contained";
+import { Address } from "@metriport/carequality-sdk/models/address";
+
+export type XCUrls = {
+  urlXCPD: string;
+  urlDQ?: string;
+  urlDR?: string;
+};
+
+export function getUrls(contained: Contained): XCUrls | undefined {
+  const endpointMap: Record<string, string> = {};
+
+  contained?.forEach(endpoint => {
+    const ext = endpoint?.Endpoint.extension.extension.find(ext => ext.url === "Transaction");
+    const type = ext?.valueString?.value;
+
+    if (type && endpoint && endpoint.Endpoint.address && endpoint.Endpoint.address.value) {
+      endpointMap[type] = endpoint.Endpoint.address.value;
+    }
+  });
+
+  const urlXCPD = endpointMap["XCPD ITI-55"];
+
+  if (!urlXCPD) return;
+
+  const urls: XCUrls = {
+    urlXCPD,
+  };
+
+  if (endpointMap["XCA ITI-38"]) {
+    urls.urlDQ = endpointMap["XCA ITI-38"];
+  }
+  if (endpointMap["XCA ITI-39"]) {
+    urls.urlDR = endpointMap["XCA ITI-39"];
+  }
+
+  return urls;
+}
+
+export function getCoordinates(
+  address: Address[] | undefined
+): { latitude: string; longitude: string } | undefined {
+  if (!address) return;
+  const coordinates = address.flatMap(a => {
+    if (a.extension?.url === "OrgPosition") {
+      const position = a.extension?.valueCodeableConcept?.coding?.value?.position;
+      if (!position) return [];
+      return {
+        latitude: position.latitude.value,
+        longitude: position.longitude.value,
+      };
+    }
+  })[0];
```

Comment:
If we can have multiple addresses, why are we returning only the first set of coordinates?

Reply:
@leite08 - this may be b/c CQ uses FHIR STU3 to represent a directory entry, which has support for multiple addresses by default, but in practice, I imagine this will always be an array with a single element

we'll only really be able to confirm this by either getting access to the prod dir, or pinging Bill @ CQ to ask AFIAK - @RamilGaripov might have more info here though, not sure what the spec says?

Reply:
https://carequality.org/healthcare-directory/StructureDefinition-SequoiaOrganization.html

Address's cardinality is set to 1..* So I'm not 100% sure. But from looking at some 30+ entries on DEV, all of them had 1 address